### PR TITLE
6 get api articles id comments

### PR DESCRIPTION
**Pull request by VenturingForth**
Added GET /api/articles/:article_id/comments

Tested the following: 
200: Comments should be retrieved with the necessary properties in an array of objects
200: Comments should be sorted by most recent first.
200: Article with no comments should send empty array
400: Should catch invalid IDs in previous testing
404: Should catch valid non-existing IDs in previous testing

Update the Endpoints JSON with the relevant data.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -104,6 +104,49 @@ describe("GET /api", () => {
                     expect(body.msg).toBe("Bad request");
                 })
             })
+            describe("/comments", () => {
+                test("200: Comments should be retrieved with the necessary properties in an array of objects", () => {
+                    return request(app)
+                    .get('/api/articles/1/comments')
+                    .expect(200)
+                    .then(({body}) => {
+                        console.log(body);
```
Comment (nc-john-hutcheon) [inline __tests__/app.test.js:113]: Just remember to remove console logs before pushing code please 🙂

File: __tests__/app.test.js [hunk 2]
```diff
@@ -104,6 +104,49 @@ describe("GET /api", () => {
...
+                        expect(body.comments.length).toBe(0);
+                    })
+                })
+                test("404: Should return \"Article ID not found\" if given valid but non-existent ID", () => {
+                    return request(app)
+                    .get('/api/articles/999/comments')
+                    .expect(404)
+                    .then(({body}) => {
+                        expect(body.msg).toBe("Article ID not found")
+                    })
```
Comment (nc-john-hutcheon) [inline __tests__/app.test.js:147]: Great testing so far! Could we also have a test for if we pass in an invalid ID? E.g. api/articles/invalid-id/comments

File: controllers/articles-controllers.js [hunk 3]
```diff
@@ -12,4 +14,18 @@ module.exports.getArticleById = (req, res, next) => {
     return fetchArticleById(article_id).then((article) => {
         res.status(200).send({article})
     }).catch(next);
+}
+
+module.exports.getArticleComments = (req, res, next) => {
+    const { article_id } = req.params;
+    const checkIdExistsQuery = checkArticleIdExists(article_id);
+    const commentsQuery = fetchArticleComments(article_id);
+    const queryArray = [commentsQuery, checkIdExistsQuery];
```
Comment (nc-john-hutcheon) [inline controllers/articles-controllers.js:23]: Nicely handled 🙌

File: utils.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+const db = require("./db/connection.js");
+
+module.exports.checkArticleIdExists = (article_id) => {
+    return db.query(`SELECT * FROM articles WHERE article_id = $1`, [article_id])
```
Comment (nc-john-hutcheon) [inline utils.js:4]: beautiful return statement 😉

**Review by nc-john-hutcheon — COMMENTED** [review]
Great work Josh! I've just left a few comments including one extra test you could add please. Well done! ⭐

File: __tests__/app.test.js (hunk 1 above)
Comment (VenturingForth) [reply __tests__/app.test.js:113]: Refactored.

File: __tests__/app.test.js (hunk 2 above)
Comment (VenturingForth) [reply __tests__/app.test.js:147]: This has been refactored using existing error handles.

File: controllers/articles-controllers.js (hunk 3 above)
Comment (VenturingForth) [reply controllers/articles-controllers.js:23]: Thanks!

File: utils.js (hunk 4 above)
Comment (VenturingForth) [reply utils.js:4]: Return of the return. Thanks!

**Pull request merged**